crates/core/tedge_actors/src/builders.rs

```diff
@@ -0,0 +1,19 @@
+use crate::{DynSender, LinkError, RuntimeError, RuntimeHandle};
+use async_trait::async_trait;
+
+/// Materialize an actor instance under construction
+///
+/// Such an instance is:
+/// 1. built from some actor configuration
+/// 2. connected to other peers
+/// 3. eventually spawned into an actor.
+#[async_trait]
+pub trait ActorBuilder {
+    /// Build and spawn the actor
+    async fn spawn(self, runtime: &mut RuntimeHandle) -> Result<(), RuntimeError>;
+}
+
+/// Implemented by actor builders to connect the actors under construction
+pub trait PeerLinker<Input, Output> {
+    fn connect(&mut self, output_sender: DynSender<Output>) -> Result<DynSender<Input>, LinkError>;
```

Comment:
I agree that a complex actor like the `ConfigManagerActor` will need multiple such connections and each connection with different semantics, one for HTTP, for for MQTT and another one for FSEvents etc. That's exactly why I was wondering if a generic `connect` API like this (which could be called multiple times) is sufficient. Each "connection" comes with a semantics that it is an "HTTP connection" or "MQTT connection". So I feel that they would even be handled differently rather than treating them as just a list of `DynSender`s.

This even got highlighted in the `ConfigManager`Builder struct where a concrete `with_http_connection` API had to be added, and I was going to introduce a `with_mqtt_connection` API as well. So, I feel that such customized "connect APIs" would be easier to manage than a single `connect` API that can accept multiple `DynSender`s.

Reply:
This  `PeerLinker` trait has been removed and replaced by two traits,[`MessageBoxConnector`](https://github.com/thin-edge/thin-edge.io/pull/1622/files#diff-d6192a6d562fd29bed522731372f341b715cb6560f8def001d9860f9b6de3e11R44) and [`MessageBoxPort`](https://github.com/thin-edge/thin-edge.io/pull/1622/files#diff-d6192a6d562fd29bed522731372f341b715cb6560f8def001d9860f9b6de3e11R62), that tell how to connect the message boxes of a service provider and a service consumer.